### PR TITLE
Prevent big coin registration if less than 5 confirmation.

### PR DESCRIPTION
**Pull request by molnard**

### CODE REVIEW

**Comment by molnard** [comment]
Changed the default conf target to 6 https://github.com/zkSNACKs/CoinVerifier/issues/8#issuecomment-1371195085

**Comment by thibistaken** [comment]
Reading the code, it seems that a coin with less than 6 confirmations would get banned for 31 days. Is that true? And if not, what is the ban time such that the coin can attempt another registration once 6 conf is met? I also wonder why there is a "big coin"limit of 1 BTC. Would that mean that a coin of 0.1 BTC can bypass the 6 conf requirements to enter input registration phase?

**Comment by molnard** [comment]
> would get banned for 31 days

No. Only coins with these error codes - https://github.com/molnard/WalletWasabi/blob/prevent5conf/WalletWasabi/WabiSabi/Backend/Models/WabiSabiProtocolErrorCode.cs#L44

> what is the ban time

There is no ban time. It behaves the same as an immature coin or an unconfirmed. It is just rejected. The UTXO can try again whenever. 

> why there is a "big coin"limit of 1 BTC

This solution is far from final - this is a hotfix. Waiting 6 confirmations for every fresh coin is a no-go from business side. On the other hand, allowing hacked coins is a nogo either. Still nobody wants to see big amount of hacked coins mixed with the wallet so until CA or us find a better solution for the 6 confirmation requirement - this can be a mitigation.

File: WalletWasabi/WabiSabi/Backend/Rounds/Arena.Partial.cs [hunk 1]
```diff
@@ -391,7 +391,15 @@ public async Task<Coin> OutpointToCoinAsync(InputRegistrationRequest request, Ca
 			throw new WabiSabiProtocolException(WabiSabiProtocolErrorCode.InputImmature);
 		}
 
-		return new Coin(input, txOutResponse.TxOut);
+		var coin = new Coin(input, txOutResponse.TxOut);
+
+		if (coin.Amount >= Config.CoinVerifierRequiredConfirmationAmount && txOutResponse.Confirmations < Config.CoinVerifierRequiredConfirmation)
+		{
+			Logger.LogInfo($"Coin cannot be verified - preventing registration. Amount: {coin.Amount}/{Config.CoinVerifierRequiredConfirmationAmount} Confirmations: {txOutResponse.Confirmations / Config.CoinVerifierRequiredConfirmation}.");
```
Comment (Szpoti) [inline WalletWasabi/WabiSabi/Backend/Rounds/Arena.Partial.cs:398]:
```suggestion
			Logger.LogInfo($"Coin cannot be verified - preventing registration. Amount: {coin.Amount}/{Config.CoinVerifierRequiredConfirmationAmount} Confirmations: {txOutResponse.Confirmations} / {Config.CoinVerifierRequiredConfirmation}.");
```

**Review by Szpoti — APPROVED** [review]
ACK

File: WalletWasabi/WabiSabi/Backend/Rounds/Arena.Partial.cs (hunk 1 above)
Comment (molnard) [reply WalletWasabi/WabiSabi/Backend/Rounds/Arena.Partial.cs]: Good catch fixed

**Comment by molnard** [comment]
This is not taking the coinjoined coins into account. Refactoring - brb.

**Comment by molnard** [comment]
I had to move the GetTxOut check right before we start communication with CA. That is the place where I am aware of which coins are meant to be verified. WIP

**Comment by molnard** [comment]
This will be solved in a different way https://github.com/molnard/WalletWasabi/pull/46

**Pull request closed without merging**